benchmark/components/simd_fnt.cpp

```diff
@@ -0,0 +1,370 @@
+            this->q = static_cast<T>(65537);
+            this->word_size = 2;
+        } else {
+            throw "Wrong TypeParam for PerfSimd tests";
+        }
+
+        this->distribution =
+            std::make_unique<std::uniform_int_distribution<uint32_t>>(0, q - 1);
+
+        gf = gf::alloc<gf::Field<T>, gf::Prime<T>>(q);
+
+        n = 2;
+        size = simd::countof<T>() * MAX_VEC_LEN;
+        wk_data = std::make_unique<vec::Buffers<T>>(n, size);
+        randomize_data(*wk_data);
+    }
+
+    void randomize_data(const vec::Buffers<T>& vec, int _max = 0)
+    {
+        const T max = (_max == 0) ? q : _max;
+        std::uniform_int_distribution<T> dis(0, max - 1);
+
+        const size_t vec_n = vec.get_n();
+        const size_t vec_size = vec.get_size();
+
+        const std::vector<T*>& mem = vec.get_mem();
+        for (size_t i = 0; i < vec_n; ++i) {
+            for (size_t j = 0; j < vec_size; j++) {
+                mem[i][j] = dis(quadiron::prng());
+            }
+        }
+    }
+
+    template <typename TFunc>
+    void field_arith(benchmark::State& st, const TFunc& f)
+    {
+        const size_t vec_len = st.range(0);
+
+        const std::vector<T*>& mem = wk_data->get_mem();
+        simd::VecType* vec_x = reinterpret_cast<simd::VecType*>(mem[0]);
+        simd::VecType* vec_y = reinterpret_cast<simd::VecType*>(mem[1]);
+
+        for (auto _ : st) {
+            for (unsigned i = 0; i < vec_len; ++i) {
+                simd::VecType x = simd::load_to_reg(&vec_x[i]);
+                simd::VecType y = simd::load_to_reg(&vec_y[i]);
+
+                f(x, y);
+
+                simd::store_to_mem(&vec_x[i], x);
+            }
+        }
+
+        const size_t bytes = vec_len * simd::countof<T>() * word_size;
+        st.counters.insert({{"Vector len", vec_len},
+                            {"Throughput",
+                             benchmark::Counter(
+                                 st.iterations() * bytes,
+                                 benchmark::Counter::kIsRate,
+                                 benchmark::Counter::OneK::kIs1024)},
+                            {"Buffer size",
+                             benchmark::Counter(
+                                 bytes,
+                                 benchmark::Counter::kDefaults,
+                                 benchmark::Counter::OneK::kIs1024)}});
+    }
+
+    template <typename TFunc>
+    void butterfly_op(benchmark::State& st, const TFunc& f)
+    {
+        const size_t vec_len = st.range(0);
+        const std::vector<T*>& mem = wk_data->get_mem();
+
+        simd::VecType* vec_x = reinterpret_cast<simd::VecType*>(mem[0]);
+        simd::VecType* vec_y = reinterpret_cast<simd::VecType*>(mem[1]);
+
+        for (auto _ : st) {
+            // use a random coefficient
+            T coef = 1
+                     + this->distribution->operator()(quadiron::prng())
+                           % (this->q - 2);
```

Comment:
Should use a distribution instead of `%`.

Reply:
It's done. Thanks